src/event_emitter/mod.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Damir Jelić
+// Copyright 2020 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use crate::events::collections::all::{RoomEvent, StateEvent};
+use crate::events::collections::only::Event as NonRoomEvent;
+use crate::events::presence::PresenceEvent;
+use crate::models::Room;
+
+use tokio::sync::Mutex;
+///
+#[async_trait::async_trait]
+pub trait EventEmitter: Send + Sync {
```

Comment:
Can we please limit this for not to the callbacks that get emitted?

Reply:
Sorry I'm not sure what you mean here.

Reply:
Ah sorry, github hid one of the file changes from me, so I didn't notice that the callbacks are actually used.

Reply:
Oh like in the `AsyncClient::sync`, so this is all good then?

Reply:
All good, yeah. I thought you added callback types that are unused and not emitted.